crates/bevy_remote/src/lib.rs

```diff
@@ -0,0 +1,877 @@
+    ///
+    /// The server responds with a `BrpResponse::Query`.
+    #[serde(rename = "QUERY")]
+    Query {
+        /// The components to select.
+        data: BrpQuery,
+
+        /// An optional filter that specifies which entities to include or
+        /// exclude from the results.
+        #[serde(default)]
+        filter: BrpQueryFilter,
+    },
+
+    /// `SPAWN`: Creates a new entity with the given components and responds
+    /// with its ID.
+    ///
+    /// The server responds with a `BrpResponse::Entity`.
+    #[serde(rename = "SPAWN")]
+    Spawn {
+        /// A map from each component's *full path* to its serialized value.
+        ///
+        /// These components will be added to the entity.
+        ///
+        /// Note that the keys of the map must be the *full* type paths: e.g.
+        /// `bevy_transform::components::transform::Transform`, not just
+        /// `Transform`.
+        components: HashMap<String, Value>,
+    },
+
+    /// `DESTROY`: Given an ID, despawns the entity with that ID.
+    ///
+    /// The server responds with a `BrpResponse::Ok`.
+    #[serde(rename = "DESTROY")]
+    Destroy {
+        /// The ID of the entity to despawn.
+        entity: Entity,
+    },
+
+    /// `REMOVE`: Deletes one or more components from an entity.
+    ///
+    /// The server responds with a `BrpResponse::Ok`.
+    #[serde(rename = "REMOVE")]
+    Remove {
+        /// The ID of the entity from which components are to be removed.
+        entity: Entity,
+
+        /// The *full paths* of the component types that are to be removed from
+        /// the entity.
+        ///
+        /// Note that these strings must consist of the *full* type paths: e.g.
+        /// `bevy_transform::components::transform::Transform`, not just
+        /// `Transform`.
+        components: Vec<String>,
+    },
+
+    /// `INSERT`: Adds one or more components to an entity.
+    ///
+    /// The server responds with a `BrpResponse::Ok`.
+    #[serde(rename = "INSERT")]
+    Insert {
+        /// The ID of the entity that components are to be added to.
+        entity: Entity,
+
+        /// A map from each component's *full path* to its serialized value.
+        ///
+        /// These components will be added to the entity.
+        ///
+        /// Note that the keys of the map must be the *full* type paths: e.g.
+        /// `bevy_transform::components::transform::Transform`, not just
+        /// `Transform`.
+        components: HashMap<String, Value>,
+    },
+
+    /// `REPARENT`: Changes the parent of an entity.
+    ///
+    /// The server responds with a `BrpResponse::Ok`.
+    #[serde(rename = "REPARENT")]
+    Reparent {
+        /// The IDs of the entities that are to become the new children of the
+        /// `parent`.
+        entities: Vec<Entity>,
+
+        /// The IDs of the entity that will become the new parent of the
+        /// `entities`.
+        parent: Entity,
+    },
```

Comment:
I feel like `parent` should be `Option<Entity>` to avoid the need for a future `ORPHAN` command as well.

Reply:
This is done now.